modules/images-configuration-registry-mirror-configuring.adoc

```diff
@@ -0,0 +1,274 @@
++
+.Example output
+[source,terminal]
+----
+unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
+short-name-mode = ""
+
+[[registry]]
+  prefix = ""
+  location = "registry.access.redhat.com/ubi9/ubi-minimal" <1>
+
+  [[registry.mirror]]
+    location = "example.io/example/ubi-minimal" <2>
+    pull-from-mirror = "digest-only" <3>
+
+  [[registry.mirror]]
+    location = "example.com/example/ubi-minimal"
+    pull-from-mirror = "digest-only"
+
+[[registry]]
+  prefix = ""
+  location = "registry.example.com"
+
+  [[registry.mirror]]
+    location = "mirror.example.net/registry-example-com"
+    pull-from-mirror = "digest-only"
+
+[[registry]]
+  prefix = ""
+  location = "registry.example.com/example"
+
+  [[registry.mirror]]
+    location = "mirror.example.net"
+    pull-from-mirror = "digest-only"
+
+[[registry]]
+  prefix = ""
+  location = "registry.example.com/example/myimage"
+
+  [[registry.mirror]]
+    location = "mirror.example.net/image"
+    pull-from-mirror = "digest-only"
+
+[[registry]]
+  prefix = ""
+  location = "registry.redhat.io"
+
+  [[registry.mirror]]
+    location = "mirror.example.com"
+    pull-from-mirror = "digest-only"
+
+[[registry]]
+  prefix = ""
+  location = "registry.redhat.io/openshift4"
+
+  [[registry.mirror]]
+    location = "mirror.example.com/redhat"
+    pull-from-mirror = "digest-only"
+[[registry]]
+  prefix = ""
+  location = "registry.access.redhat.com/ubi9/ubi-minimal"
+  blocked = true <4>
+
+  [[registry.mirror]]
+    location = "example.io/example/ubi-minimal-tag"
+    pull-from-mirror = "tag-only" <5>
+----
+<1> Indicates the repository that is referred to in a pull spec.
+<2> Indicates the mirror for that repository.
+<3> Indicates that the image pull from the mirror is a digest reference image.
+<4> Indicates that the `NeverContactSource` parameter is set for this repository.
+<5> Indicates that the image pull from the mirror is a tag reference image.
+
+.. Pull an image to the node from the source and check if it is resolved by the mirror.
++
+[source,terminal]
+----
+sh-4.2# podman pull --log-level=debug registry.access.redhat.com/ubi9/ubi-minimal@sha256:5cf...
+----
+
+.Troubleshooting repository mirroring
+
+If the repository mirroring procedure does not work as described, use the following information about how repository mirroring works to help troubleshoot the problem.
+
+* The first working mirror is used to supply the pulled image.
+* The main registry is only used if no other mirror works.
+* From the system context, the `Insecure` flags are used as fallback.
+* The format of the `/etc/containers/registries.conf` file has changed recently. It is now version 2 and in TOML format.
+* You cannot add the same repository to both an `ImageDigestMirrorSet` and an `ImageTagMirrorSet` object.
```

Comment:
> Is this true? The migration doc says IDMS and ITMS can use the same registry.

It's true. The same mirror can stay in `ImageDigestMirrorSet` and an `ImageTagMirrorSet`. Whether the mirror will be used during the pull depends on how the pull spec is specified. We can remove this line from the doc.